### PR TITLE
[Heartbeat] Incorporate factory metadata for autodiscover

### PR DESCRIPTION
**Pull request by andrewvc**
Heartbeat factories get metadata from autodiscover and other sources.

This change automatically adds that data to events keeping heartbeat behavior in-line with other beats.

### CODE REVIEW

**Comment by andrewvc** [comment]
@ruflin my original thinking was that the additional fields here might be considered 'breaking', but I no longer believe that. I think that was backwards. I've added the `needs_backport` label. LMK if you think that isn't deserved. I think it should make it into 6.7 if possible.

**Comment by andrewvc** [comment]
@ruflin added changelog entry (and fixed a previous one that was in the wrong spot)

**Comment by ruflin** [comment]
jenkins, test this

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -112,6 +112,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 *Heartbeat*
 
 - Made monitors.d configuration part of the default config. {pull}9004[9004]
+- Fixed rare issue where TLS connections to endpoints with x509 certificates missing either notBefore or notAfter would cause the check to fail with a stacktrace.  {pull}9566[9566]
```
Comment (ruflin) [inline CHANGELOG.next.asciidoc]: Does not seem to belong to this PR?

Comment (andrewvc) [reply CHANGELOG.next.asciidoc]: Technically no, but I moved it since it was in the wrong spot. Should I make a new PR?

Comment (ruflin) [reply CHANGELOG.next.asciidoc]: Just realised now that you moved it only and was not added. All good.

File: heartbeat/monitors/task.go [hunk 2]
```diff
@@ -133,10 +133,15 @@ func (t *configuredJob) makeSchedulerTaskFunc() scheduler.TaskFunc {
...
+	fields := common.MapStr{"event": common.MapStr{"dataset": "uptime"}}
+	if t.monitor.factoryMetadata != nil {
+		fields.DeepUpdate(t.monitor.factoryMetadata.Get())
+	}
+
 	t.client, err = t.monitor.pipelineConnector.ConnectWith(beat.ClientConfig{
 		EventMetadata: t.config.EventMetadata,
 		Processor:     t.processors,
-		Fields:        common.MapStr{"event": common.MapStr{"dataset": "uptime"}},
+		Fields:        fields,
```
Comment (exekias) [inline heartbeat/monitors/task.go]: there is a `DynamicFields` here in `ClientConfig`, were you can pass t.monitor.factoryMetadata directly, no need for that `DeepUpdate`
